ember_debug/models/profile-manager.js

```diff
@@ -5,6 +5,63 @@ const {
   run: { later, scheduleOnce },
 } = Ember;
 
+function getEdges(first, last, closest) {
+  let start = null;
+  let end = null;
+  for (let i = 0; i < closest.length; i++) {
+    if (closest.item(i) === first.node) start = i;
+    else if (closest.item(i) === last.node) end = i;
+  }
+  return [start, end];
+}
+
+function getUnfilteredRoots(first, last, closest) {
+  if (first.node === last.node) return [first.node];
+
+  const roots = [];
+
+  const [start, end] = getEdges(first, last, closest);
+
+  if (start === null || end === null) return [];
+
+  for (let i = start; i <= end; i++) roots.push(closest.item(i));
+
+  return roots;
+}
+
+function findRoots({ first, last, parent }) {
+  const closest = parent.childNodes;
+
+  const roots = getUnfilteredRoots(first, last, closest);
+
+  return roots.filter((el) => el?.nodeType === 1);
+}
+
+function makeHighlight() {
+  const node = document.createElement('div');
+  node.setAttribute('role', 'presentation');
+  node.setAttribute('class', 'ember-inspector-render-highlight');
+  return node;
+}
+function insertHTML() {
+  return document.body.appendChild(makeHighlight());
```

Comment:
let's get makeHighlight() result as function argument here (to store DOM reference in object for destroying)